### PR TITLE
BUILD/PGCC: set of fixes to build by pgcc compiler

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed set of mixes int<->enum data types

### CODE REVIEW

**Comment by shamisp** [comment]
What PGCC version is this ?

**Comment by hoopoepg** [comment]
Rel Dev-r194825 LLVM 64-bit

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -33,6 +33,13 @@
 #include <string.h>
 
 
+/**
+ * Enumerate all memory types
+ */
+#define ucp_ep_for_each_memory_type(_x) \
+    for (_x = UCS_MEMORY_TYPE_HOST; _x < UCS_MEMORY_TYPE_LAST; _x++)
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c:40]: move to ucs

File: src/ucp/wireup/ep_match.c [hunk 2]
```diff
@@ -80,20 +82,24 @@ void ucp_ep_match_insert(ucp_worker_h worker, ucp_ep_h ep, uint64_t dest_uuid,
...
+                               ucp_ep_match_conn_sn_t conn_sn,
+                               ucs_conn_match_queue_type_t conn_queue_type)
 {
-    ucp_ep_flags_t UCS_V_UNUSED exp_ep_flags = UCP_EP_FLAG_ON_MATCH_CTX |
-                                              (is_exp ?
-                                               0 : UCP_EP_FLAG_REMOTE_ID);
+    ucp_ep_flags_t UCS_V_UNUSED exp_ep_flags =
+        UCP_EP_FLAG_ON_MATCH_CTX |
+        ((conn_queue_type == UCS_CONN_MATCH_QUEUE_EXP) ?
+         0 : UCP_EP_FLAG_REMOTE_ID);
```
Comment (yosefe) [inline src/ucp/wireup/ep_match.c:96]: use if, too complicated

File: src/ucp/wireup/select.c [hunk 3]
```diff
@@ -538,7 +538,8 @@ ucp_wireup_add_lane_desc(const ucp_wireup_select_info_t *select_info,
     lane_desc->proxy_lane   = proxy_lane;
     lane_desc->dst_md_index = dst_md_index;
     lane_desc->lane_types   = UCS_BIT(lane_type);
-    for (lane_type_iter = 0; lane_type_iter < UCP_LANE_TYPE_LAST;
+    UCS_STATIC_ASSERT(0 == UCP_LANE_TYPE_AM);
+    for (lane_type_iter = UCP_LANE_TYPE_AM; lane_type_iter < UCP_LANE_TYPE_LAST;
```
Comment (yosefe) [inline src/ucp/wireup/select.c:542]: maybe for such enums, add FIRST constant?

File: src/uct/ib/base/ib_iface.h [hunk 4]
```diff
@@ -21,7 +21,7 @@
 #define UCT_IB_IFACE_NULL_RES_DOMAIN_KEY   0u
 #define UCT_IB_MAX_ATOMIC_SIZE             sizeof(uint64_t)
 #define UCT_IB_ADDRESS_INVALID_GID_INDEX   UINT8_MAX
-#define UCT_IB_ADDRESS_INVALID_PATH_MTU    0
+#define UCT_IB_ADDRESS_INVALID_PATH_MTU    (enum ibv_mtu)0
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.h:24]: ()

File: src/ucp/wireup/ep_match.c [hunk 5]
```diff
@@ -110,13 +116,17 @@ ucp_ep_h ucp_ep_match_retrieve(ucp_worker_h worker, uint64_t dest_uuid,
...
     if (!(ep->flags & UCP_EP_FLAG_ON_MATCH_CTX)) {
         return;
     }
 
+    conn_queue_type = (ep->flags & UCP_EP_FLAG_REMOTE_ID) ?
+                      UCS_CONN_MATCH_QUEUE_UNEXP : UCS_CONN_MATCH_QUEUE_EXP;
     ucs_conn_match_remove_elem(&worker->conn_match_ctx,
                                &ucp_ep_ext_gen(ep)->ep_match.conn_match,
-                               !(ep->flags & UCP_EP_FLAG_REMOTE_ID));
+                               conn_queue_type);
```
Comment (dmitrygx) [inline src/ucp/wireup/ep_match.c:129]: can avoid `conn_queue_type` variable and pass the result of ternary operator directly

File: src/ucp/core/ucp_ep.c (hunk 1 above)
Comment (dmitrygx) [reply src/ucp/core/ucp_ep.c:40]: also, wrap `_x` by `()`

**Review by dmitrygx — CHANGES_REQUESTED** [review]

File: src/ucp/core/ucp_ep.c (hunk 1 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_ep.c]: moved & wrapped

File: src/ucp/wireup/ep_match.c (hunk 2 above)
Comment (hoopoepg) [reply src/ucp/wireup/ep_match.c]: ok, used ```if```

File: src/ucp/wireup/ep_match.c (hunk 5 above)
Comment (hoopoepg) [reply src/ucp/wireup/ep_match.c]: ok, done

File: src/ucp/wireup/select.c (hunk 3 above)
Comment (hoopoepg) [reply src/ucp/wireup/select.c]: ok, added FIRST enum value

File: src/uct/ib/base/ib_iface.h (hunk 4 above)
Comment (hoopoepg) [reply src/uct/ib/base/ib_iface.h]: wrapped into ()

File: src/ucs/config/types.h [hunk 6]
```diff
@@ -110,6 +110,14 @@ typedef enum {
 
 typedef UCS_CONFIG_STRING_ARRAY_FIELD(names) ucs_config_names_array_t;
 
+
+/**
+ * Enumerate all memory types
+ */
+#define ucs_sys_for_each_memory_type(_x) \
```
Comment (yosefe) [inline src/ucs/config/types.h:117]: ucs_memory_type_for_each
pls put in ucs/memory/memory_type.h

**Review by yosefe — APPROVED** [review]

**Comment by hoopoepg** [comment]
I will squash after tests are completed

**Pull request merged**